packages/electron/src/telemetry/electron-telemetry.ts

```diff
@@ -11,57 +11,82 @@ import {
 } from "@porting-assistant/react/src/models/project";
 import { SolutionDetails } from "@porting-assistant/react/src/models/solution";
 import crypto from "crypto";
-import log, { LogMessage, LevelOption } from "electron-log";
+import fs from "fs";
+import log, { LogMessage, LevelOption, info } from "electron-log";
 import { Connection } from "electron-cgi/connection";
 import { putMetricData } from "./electron-metrics";
 import { localStore } from "../preload-localStore";
+import path from "path";
+import winston from "winston";
+import DailyRotateFile from "winston-daily-rotate-file";
+
+const logFileName = "porting-assistant-%DATE%";
+
+const BACKEND_LOG = "portingAssistant-backend-logs";
+const ELECTRON_LOG = "electron-logs";
+const REACT_ERROR = "react-errors";
+
+const dirName = path.join(app.getPath("userData"), "telemetry-logs");
+
+if (!fs.existsSync(dirName))
+  fs.mkdir(dirName, (err) => {
+    console.log("Telemetry Directory Creation Failed.");
+  });
+
+var winstonTransports = [
+  new DailyRotateFile({
+    datePattern: "YYYY-MM-DD",
+    extension: ".log",
+    filename: logFileName,
+    dirname: dirName,
+    maxSize: 1024 * 1024,
```

Comment:
maxSize and maxFiles might lead to missing electron logs, since the uploader is now a timed event. In a scenario where these thresholds are crossed within our timed event limits, log would be rotated and we would miss these log files. Have a look at how this possibility can be minimized or removed altogether (maybe by only having daily/time based rotation).